config/checkstyle.xml

```diff
@@ -186,7 +186,7 @@
             <property name="allowedAbbreviationLength" value="1"/>
         </module> -->
         <module name="OverloadMethodsDeclarationOrder" />
-        <module name="VariableDeclarationUsageDistance" />
+        <!-- <module name="VariableDeclarationUsageDistance" /> -->
```

Comment:
This is potentially contraversial so happy to reenable the check and fix the problems if people disagree with me here. 

The reason for disabling the check is that I think it can make writing tests with setup data hard. For instance:

```java
String testData1 = // setup test data
String testData2 = // setup test data
String testData3 = // setup tests data

testObject.doAThing(testData1);
testObject.doAThing(testData2);
testObject.doSomethingElse();

assertThat(testObject.getSomething(), notNullValue());
assertThat(testObject.getSomething().anotherThing();, equalTo(testData3));
```

This is a pretty common pattern for a test - we're setting up our inputs and outputs, carrying out the steps we're testing and then making an assertion on something we set up at the beginning of the test. This can make tests more readable and also helps if you need to mutate input/output values. This would fail our `checkstyle` build however as the `testData3` declaration would be too far away from it's use.

Reply:
I agree that this split is reasonable for tests. I do't feel strongly one way or the other but I will say that I have had that checkstyle rule fail on code I was writing and was glad for it. Maybe it's something that would have been flagged in review anyway.